### PR TITLE
Lint fixes

### LINKED ISSUE #171: Fix Lint errors/warnings
There are 407 currently.
It would be good to solve them little by little.

### PR DESCRIPTION
**Pull request by betabear26**
**Description (required)**
Lint fixes for multiple packages.

Fixes #171  Fix Lint Errors and warnings.

What changes did you make and why?

Fixed multiple lint warnings. Many requires in depth understanding of code, will be doing it subsequently.

**Tests performed (required)**

Tested ProdDebug on OnePlus 6 with API level 28.

Need help? See https://support.google.com/android/answer/9075928

---

_Note: Please ensure that you have read CONTRIBUTING.md if this is your first pull request._

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java [hunk 1]
```diff
@@ -120,6 +120,7 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
         if (mediaDetails == null || !mediaDetails.isVisible()) {
             // set isFeaturedImage true for featured images, to include author field on media detail
             mediaDetails = new MediaDetailPagerFragment(false, true);
+            //FIXME: Is this even required. Check it once.
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java]: Which line are you referring to here? `supportFragmentManager` is used twice below.

File: app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java [hunk 2]
```diff
@@ -112,6 +112,7 @@ public void onItemClick(AdapterView<?> adapterView, View view, int i, long l) {
         if (mediaDetails == null || !mediaDetails.isVisible()) {
             // set isFeaturedImage true for featured images, to include author field on media detail
             mediaDetails = new MediaDetailPagerFragment(false, true);
+            //FIXME: check if it is even required.
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java]: What is `it`? Have a feeling this is similar to `CategoryDetailsActivity.java`

File: app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java [hunk 3]
```diff
@@ -220,9 +220,10 @@ private String licenseTemplateFor(String license) {
                 return "{{self|cc-by-3.0}}";
             case Prefs.Licenses.CC_BY_SA:
                 return "{{self|cc-by-sa-3.0}}";
-        }
+                default:
+                    throw new RuntimeException("Unrecognized license value: " + license);
 
-        throw new RuntimeException("Unrecognized license value: " + license);
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/contributions/Contribution.java]: I feel it's easier just like this. However if you are going to put it into switch can you decrease indentation of lines 223 and 224 to match the rest.

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListAdapter.java [hunk 4]
```diff
@@ -64,6 +64,8 @@ public void bindView(View view, Context context, Cursor cursor) {
                 views.stateView.setText(R.string.contribution_state_failed);
                 views.progressView.setVisibility(View.GONE);
                 break;
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListAdapter.java]: Also can remove this break

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListAdapter.java [hunk 5]
```diff
@@ -64,6 +64,8 @@ public void bindView(View view, Context context, Cursor cursor) {
...
+                default:
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListAdapter.java]: No point in this?

File: app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java [hunk 6]
```diff
@@ -237,28 +237,27 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
...
-            }
             break;
-            case 3: {
+            case 3:
                 if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                     Timber.d("Call controller.startCameraCapture()");
                     controller.startCameraCapture();
                 }
-            }
+                default:
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/contributions/ContributionsListFragment.java]: Good catch. Needs to decrease indentation

File: app/src/main/java/fr/free/nrw/commons/delete/DeleteTask.java [hunk 7]
```diff
@@ -147,6 +147,8 @@ protected void onProgressUpdate (Integer... values){
             case 5:
                 message = "Done";
                 break;
+                default:
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/delete/DeleteTask.java]: Unnecessary? Also remove break on line 149?

File: app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java [hunk 8]
```diff
@@ -27,7 +27,7 @@
         HasBroadcastReceiverInjector,
         HasContentProviderInjector {
 
-    private static ApplicationlessInjection instance = null;
+    private static volatile ApplicationlessInjection instance = null;
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java]: Necessary? Feel this is more than a lint issue.

File: app/src/main/java/fr/free/nrw/commons/di/CommonsApplicationModule.java [hunk 9]
```diff
@@ -40,7 +40,7 @@ public Context providesApplicationContext() {
 
     @Provides
     public AccountUtil providesAccountUtil(Context context) {
-        return new AccountUtil(context);
+        return new AccountUtil();
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/di/CommonsApplicationModule.java]: Feel this is more than a lint issue

File: app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java [hunk 10]
```diff
@@ -163,14 +163,14 @@ public void addImagesToList(String query) {
...
-        if (mediaList.size()!=0){
-            if (!queryList.get(queryList.size()-1).getFilename().equals(mediaList.get(mediaList.size()-1).getFilename())) {
-                queryList.addAll(mediaList);
-                imagesAdapter.addAll(mediaList);
-                imagesAdapter.notifyDataSetChanged();
-                ((SearchActivity)getContext()).viewPagerNotifyDataSetChanged();
-            }
+        if (!mediaList.isEmpty() &&
+                (!queryList.get(queryList.size()-1).getFilename().equals(mediaList.get(mediaList.size()-1).getFilename()))) {
+            queryList.addAll(mediaList);
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: Increase indentation of lines 168-171

File: app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java [hunk 11]
```diff
@@ -388,11 +387,7 @@ public void onDeleteButtonClicked(){
         input.addTextChangedListener(new TextWatcher() {
             private void handleText() {
                 final Button okButton = d.getButton(AlertDialog.BUTTON_POSITIVE);
-                if (input.getText().length() == 0) {
-                    okButton.setEnabled(false);
-                } else {
-                    okButton.setEnabled(true);
-                }
+                okButton.setEnabled(input.getText().length() == 0);
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/media/MediaDetailFragment.java]: Think this should be `!=`

File: app/src/main/java/fr/free/nrw/commons/modifications/CategoryModifier.java [hunk 12]
```diff
@@ -6,9 +6,9 @@
 
 public class CategoryModifier extends PageModifier {
 
-    public static String PARAM_CATEGORIES = "categories";
+    public static final String PARAM_CATEGORIES = "categories";
 
-    public static String MODIFIER_NAME = "CategoriesModifier";
+    public static final  String MODIFIER_NAME = "CategoriesModifier";
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/modifications/CategoryModifier.java]:
```suggestion
    public static final String MODIFIER_NAME = "CategoriesModifier";
```

**Comment by domdomegg** [comment]
Thank you for the many lint fixes. I've gone through and added my suggestions :)

File: app/src/main/java/fr/free/nrw/commons/achievements/BitmapUtils.java [hunk 13]
```diff
@@ -28,11 +34,11 @@ public static BitmapDrawable writeOnDrawable(Bitmap bm, String text, Context con
         Paint paint = new Paint(Paint.ANTI_ALIAS_FLAG);
         paint.setStyle(Paint.Style.FILL);
         paint.setColor(Color.WHITE);
-        paint.setTextSize(Math.round(canvas.getHeight()/2));
```
[the remark this answers is not in the document]
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/achievements/BitmapUtils.java]: canvas.getHeight() returns an integer. Math.round() is a redundancy while doing an integer division by 2.
eg. 41/2 and Math.round(41/2) will output the same result.

File: app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java [hunk 14]
```diff
@@ -297,12 +297,9 @@ private void handleOtherResults(String result) {
         if (result.equals("NetworkFailure")) {
             // Matches NetworkFailure which is created by the doInBackground method
             showMessageAndCancelDialog(R.string.login_failed_network);
-        } else if (result.toLowerCase(Locale.getDefault()).contains("nosuchuser".toLowerCase()) || result.toLowerCase().contains("noname".toLowerCase())) {
-            // Matches nosuchuser, nosuchusershort, noname
-            showMessageAndCancelDialog(R.string.login_failed_wrong_credentials);
-            emptySensitiveEditFields();
-        } else if (result.toLowerCase(Locale.getDefault()).contains("wrongpassword".toLowerCase())) {
-            // Matches wrongpassword, wrongpasswordempty
+        } else if (result.toLowerCase(Locale.getDefault()).contains("nosuchuser".toLowerCase()) || result.toLowerCase().contains("noname".toLowerCase())
```
[the remark this answers is not in the document]
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java]: Agreed, could be written in a cleaner way.

File: app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java [hunk 15]
```diff
@@ -358,8 +355,9 @@ public boolean onOptionsItemSelected(MenuItem item) {
             case android.R.id.home:
                 NavUtils.navigateUpFromSameTask(this);
                 return true;
+                default:
+                    return super.onOptionsItemSelected(item);
         }
-        return super.onOptionsItemSelected(item);
```
[the remark this answers is not in the document]
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/auth/LoginActivity.java]: There are a lot of places in code where switch could be replaced with if. Will do it eventually.

File: app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java (hunk 1 above)
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/category/CategoryDetailsActivity.java]: This wasn't supposed to be here. It was just a point for me to check. Will remove.

File: app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java (hunk 2 above)
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/category/CategoryImagesActivity.java]: This wasn't supposed to be here. It was just a point for me to check. Will remove.

File: app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java (hunk 8 above)
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java]: Double checked locking can result in insidious bugs while being used by multiple threads. Please refer the following link for a detailed explanation.

https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java

File: app/src/main/java/fr/free/nrw/commons/di/CommonsApplicationModule.java (hunk 9 above)
Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/di/CommonsApplicationModule.java]: The private field context in AccountUtil class was an unused field. 
```
private final Context context;

    public AccountUtil(Context context) {
        this.context = context;
    }
```

The constructor here is being used to set the context, but that instance of context isn't really being used. Other methods are static with context being passed as a parameter. I've removed the constructor so it uses the default constructor now.

File: app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java (hunk 8 above)
Comment (misaochan) [reply app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java]: I don't think @domdomegg was saying that this change is wrong, just that it should be separated into a different PR. :) When we modularize PRs, it is easier to check which PR caused a specific bug, and to revert if necessary.

Comment (betabear26) [reply app/src/main/java/fr/free/nrw/commons/di/ApplicationlessInjection.java]: Aha. That makes sense. I'll group up and push stuff like this in another PR then.

**Comment by domdomegg** [comment]
Going to close for now as there's lots of merge conflicts and changes requested, and it's been over a month since any activity. Please reopen if updated.

**Pull request closed without merging**